ingests/faostat.py

```diff
@@ -39,20 +41,30 @@ class FAODataset:
     source_name: str = "Food and Agriculture Organization of the United Nations"
     _extra_metadata = {}
 
-    def __init__(self, dataset_metadata: dict):
+    def __init__(self, dataset_metadata: dict, dataset_server_metadata: dict):
```

Comment:
A minor suggestion here, I think it is cleaner if could this object with just one argument

```python
FAODataset(dataset_metadata: dict)
```

We could load the `dataset_server_metadata` using an internal class method, as the link required to download it can be found in the `dataset_metadata` object.

That is, the class method

```python
def _load_dataset_server_metadata(self, dataset_url: str) -> dict:
    # Fetch only header of the dataset file on the server, which contains additional metadata.
    head_request = requests.head(dataset_url)
    dataset_header = head_request.headers
    return cast(dict, dataset_header)
```

which could be called within the constructor:

```python
def __init__(self, dataset_metadata: dict, dataset_server_metadata: dict):
        """[summary]

        Args:
            dataset_metadata (dict): Dataset raw metadata.
            dataset_server_metadata (dict): Metadata of the dataset file in the server (which contains additional
                metadata like last modification date).
        """
        self._dataset_metadata = dataset_metadata
        self._dataset_server_metadata = self._load_dataset_server_metadata(
            dataset_url=description["FileLocation"]
        )
```

Reply:
Done in [my last commit](https://github.com/owid/walden/pull/27/commits/d9585cf2a6ebb9d68407232ba584bbca4dba3bfd).